plugins/outputs/azure_data_explorer/azure_data_explorer.go

```diff
@@ -75,22 +69,38 @@ func (adx *AzureDataExplorer) Connect() error {
 		Authorizer: authorizer,
 	}
 	client, err := kusto.New(adx.Endpoint, authorization)
-
+	adx.Log.Debug("Connect : Client initialized successfully")
 	if err != nil {
 		return err
 	}
 	adx.client = client
-	adx.ingesters = make(map[string]localIngestor)
-	adx.createIngestor = createRealIngestor
-
+	adx.ingestors = make(map[string]ingest.Ingestor)
 	return nil
 }
 
+// Clean up and close the ingestor
 func (adx *AzureDataExplorer) Close() error {
+	var errs []error
+	for _, v := range adx.ingestors {
+		if inerr := v.Close(); inerr != nil {
+			// accumulate errors while closing ingestors
+			errs = append(errs, inerr)
+		}
```

Comment:
Please use `err` for errors
```suggestion
		if err := v.Close(); err != nil {
			// accumulate errors while closing ingestors
			errs = append(errs, err)
		}
```